pkg/arvo/sys/zuse.hoon

```diff
@@ -4386,12 +4386,15 @@
         ~/  %decompress-point
         |=  dat=@
         ^-  pont
-        =+  x=(end 3 w a)
-        =+  y=:(add (pow x 3) (mul a x) b)
+        =+  x=(end 3 w dat)
+        =+  y=(exp.p (div +(^p) 4) :(sum.p (exp.p 3 x) (pro.p a x) b))
         =+  s=(rsh 3 32 dat)
+        ::  check parity
+        ::
+        =?  y  !=((sub s 2) (mod y 2))
+          (sub ^p y)
         :-  x
```

Comment:
These lines can be cleaned up a bit:
```hoon
~|  [`@ux`s `@ux`dat]
?>  |(=(2 s) =(3 s))
[x y]
```
It might be better to assert about `s` just after assigning it, though.